app/queries/servers/team.test.ts

```diff
@@ -1313,4 +1324,56 @@ describe('Team Queries', () => {
             });
         });
     });
+
+    describe('observeTeamLastChannelId', () => {
+        it('should return undefined if no team channel history exists', (done) => {
+            observeTeamLastChannelId(database, teamId).subscribe((channelId) => {
+                expect(channelId).toBeUndefined();
+                done();
+            });
+        });
+
+        it('should return the last channel id from team channel history', async () => {
+            const channelId = 'channel1';
+            await operator.handleTeamChannelHistory({
+                teamChannelHistories: [{
+                    id: teamId,
+                    channel_ids: [channelId],
+                }],
+                prepareRecordsOnly: false,
+            });
+
+            observeTeamLastChannelId(database, teamId).subscribe((lastChannelId) => {
+                expect(lastChannelId).toBe(channelId);
+            });
```

Comment:
Almost sure the test is finishing before the `expect` is called, so changing `channelID` for anything (like 'foo') will still make the test pass. I think I had some example of how to properly test this in this PR: https://github.com/mattermost/mattermost-mobile/pull/8624/files